TravelGenie/TravelGenie/Presentation/PopUp/View/PopUpViewController.swift

```diff
@@ -43,27 +46,111 @@ final class PopUpViewController: UIViewController {
+    }
+    
+    override func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+        removeObservers()
+    }
+    
+    override func touchesBegan(_ touches: Set<UITouch>, with event: UIEvent?) {
+        feedbackContentView.endFeedbackTextViewEditing()
     }
     
     // MARK: Private
     
     private func configureSubviews() {
-        view.backgroundColor = .black.withAlphaComponent(0.2)
+        view.backgroundColor = .black.withAlphaComponent(0.6)
+        configureFeedbackContentView()
+    }
+    
+    private func configureFeedbackContentView() {
+        feedbackContentView.delegate = self
+        feedbackContentView.textViewDelegate = self
+        feedbackContentView.isHidden = true
     }
     
     private func configureHierarchy() {
-        view.addSubview(contentView)
+        [endChatContentView, feedbackContentView].forEach { view.addSubview($0) }
     }
     
-    private func configureContentViewLayout(_ contentView: PopUpContentView) {
-        contentView.translatesAutoresizingMaskIntoConstraints = false
+    private func configureLayout() {
+        endChatContentView.translatesAutoresizingMaskIntoConstraints = false
         NSLayoutConstraint.activate([
-            contentView.widthAnchor.constraint(equalToConstant: 351),
-            contentView.centerXAnchor.constraint(equalTo: view.centerXAnchor),
-            contentView.centerYAnchor.constraint(equalTo: view.centerYAnchor),
+            endChatContentView.widthAnchor.constraint(equalToConstant: 351),
+            endChatContentView.centerXAnchor.constraint(equalTo: view.centerXAnchor),
+            endChatContentView.centerYAnchor.constraint(equalTo: view.centerYAnchor),
+        ])
+        
+        feedbackContentView.translatesAutoresizingMaskIntoConstraints = false
+        NSLayoutConstraint.activate([
+            feedbackContentView.widthAnchor.constraint(equalToConstant: 351),
+            feedbackContentView.centerXAnchor.constraint(equalTo: view.centerXAnchor),
+            feedbackContentView.centerYAnchor.constraint(equalTo: view.centerYAnchor),
         ])
     }
+    
+    private func registerObservers() {
+        NotificationCenter.default.addObserver(
+            self,
+            selector: #selector(moveFeedbackContentViewUp),
+            name: UIResponder.keyboardWillShowNotification,
+            object: nil)
+        NotificationCenter.default.addObserver(
+            self,
+            selector: #selector(moveFeedbackContentViewDown),
+            name: UIResponder.keyboardWillHideNotification,
+            object: nil)
+    }
+    
+    private func removeObservers() {
+        NotificationCenter.default.removeObserver(
+            self,
+            name: UIResponder.keyboardWillShowNotification,
+            object: nil)
+        NotificationCenter.default.removeObserver(
+            self,
+            name: UIResponder.keyboardWillHideNotification,
+            object: nil)
+    }
```

Comment:
메모리그래프 확인해보니 실제로는 해당 옵저버때문에 PopUpVC가 deinit되지 않고 있더라구요.

> If your app targets iOS 9.0 and later or macOS 10.11 and later, and you used [addObserver(_:selector:name:object:)](https://developer.apple.com/documentation/foundation/notificationcenter/1415360-addobserver) to create your observer, you do not need to unregister the observer. If you forget or are unable to remove the observer, the system cleans up the next time it would have posted to it.

보내주신 링크에서 해당 부분 보면 `다음 번에 post가 일어나려고 할 때 clean up 한다`라고 되어있습니다.
따라서 해당 화면의 경우 다른 화면에서 키보드가 보일 때 해당 observer가 remove되어 그 때 `PopUpVC`도 deinit될 것으로 예상되어 수동으로 remove 해 주었습니다.

Reply:
아 addObserver한 객체가 소멸될 때 Observer가 remove된다고 알고있었는데 
알고있었는데 잘 못 알고있었네요
(다음번 post가 일어날 때, remove되는거군요)

> 메모리그래프 확인해보니 실제로는 해당 옵저버때문에 PopUpVC가 deinit되지 않고 있더라구요.

remove를 호출하지 하지않더라도 deinit 구문자체는 타는 것 같은데
메모리그래프에 남아있을 수도 있군요